### PR TITLE
feat: add matomo analytics app

### LINKED ISSUE #12136: Matomo analytics
### Is your proposal related to a problem?

Please add support for [Matomo Analytics](https://matomo.org/). It's another privacy friendly analytics platform. 

### Describe the solution you'd like

Add Matomo to the list of available analytics apps. 

### Describe alternatives you've considered

I have looked at the other analytics options available but I already self-host matomo. 

### Additional context

If there is already a way to do this, please share? 

### Requirement/Document

[How to embed the matomo tracking code](https://matomo.org/faq/new-to-piwik/how-do-i-embed-the-matomo-tracking-code-in-my-website-cms/)

### PR DESCRIPTION
**Pull request by harshithpabbati**
## What does this PR do?

This PR is to add Matomo analytics in the app store

Fixes #12136 

Loom Video: https://www.loom.com/share/24882809f4f241098a3cb84c688a2c2d?sid=ac2ba341-3011-4c33-9a79-a7e5bd96e826

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

* Add Matomo analytics application to an event type.
* Open up the event page and it should automatically track it.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by ThyMinimalDev — COMMENTED** [review]
small change to do, otherwise looking good

File: packages/app-store/matomo/components/EventTypeAppCardInterface.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <AppCard
+      app={app}
+      switchOnClick={(e) => {
+        if (!e) {
+          setEnabled(false);
+        } else {
+          setEnabled(true);
+        }
```
[the remark this answers is not in the document]
Comment (harshithpabbati) [reply packages/app-store/matomo/components/EventTypeAppCardInterface.tsx:24]: Hi @ThyMinimalDev, actually this part of the code was auto-generated using app-store CLI.

Should I still make the change?

[the remark this answers is not in the document]
Comment (ThyMinimalDev) [reply packages/app-store/matomo/components/EventTypeAppCardInterface.tsx:24]: yes please if you don't mind that would be great

[the remark this answers is not in the document]
Comment (harshithpabbati) [reply packages/app-store/matomo/components/EventTypeAppCardInterface.tsx]: Just made the change!

**Review by joeauyeung — APPROVED** [review]
LGTM. The Loom helped with reviewing this PR. Thanks for your contribution and participating in the hackathon.

**Pull request merged**